lua/terrortown/migrations/2024_02_16_23_10_favorites_migration.lua

```diff
@@ -0,0 +1,50 @@
+--- @ignore
+-- Migration file for favorites
+if SERVER then
+    return function() end
+end
+
+local oldDatabaseName = "ttt_bem_fav"
+local newDatabaseName = "ttt2_shop_favorites"
+local savingKeys = {}
+
+---
+-- Checks old favorites table and merges them into the new table
+-- @realm client
+return function()
+    local sqlCommand = "CREATE TABLE " .. newDatabaseName .. " (name TEXT PRIMARY KEY)"
+
+    if not sql.TableExists(newDatabaseName) and sql.Query(sqlCommand) == false then
+        error("[TTT2] Couldn't create sql table in migrations. Query failed.")
+    end
+
+    if not sql.TableExists(oldDatabaseName) then
+        return
+    end
+
+    local retrieveOldSqlDataCommand = "SELECT * FROM ttt_bem_fav"
+    local oldFavoriteEquipments = sql.Query(retrieveOldSqlDataCommand)
+
+    if oldFavoriteEquipments == false then
+        return
+    end
```

Comment:
shouldn't we at least log a dev message that old equipment favourites could not be accessed?

additionally we would need to return on `nil` as otherwise the `pairs` call in the for loop below will fail